dialoptions.go

```diff
@@ -70,6 +76,8 @@ type DialOption interface {
 	apply(*dialOptions)
 }
 
+var defaultDialOption []DialOption
```

Comment:
Nit: `defaultDialOptions`

Reply:
Let's not use the word "default" here, then, this is too confusing.  How about "extra" or "additional" or "injected" or "plugin"instead?  Or if you have any better ideas, I'm happy to hear them.